File: server/ExpenseTracker/Models/Expense.cs
namespace ExpenseTracker.Models
{
    public class Expense : BaseEntity
    {
        public Guid UserId { get; set; } //fk
        public User? User { get; set; } //navigation proepty One
        public string? Title { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
        public Guid CategoryId { get; set; }
        public Category? Category { get; set; }
        public string? Notes { get; set; }
        //public string? AttachmentUrl { get; set; }
        public Guid BudgetId { get; set; }
        public Budget? Budget { get; set; }
    }
}

File: client/src/Interfaces/Expense.ts
import { User } from './User';
import { Category } from './Category';
import { Budget } from './Budget';

export interface Expense {
  id: string; // Assuming BaseEntity has an ID
//   createdAt?: Date; // Assuming BaseEntity has timestamps
//   updatedAt?: Date; // Assuming BaseEntity has timestamps
  userId: string;
  user: User;
  title: string;
  amount: number;
  date: Date;
  categoryId: string;
  category: Category;
  notes?: string;
  // attachmentUrl?: string;
  budgetId: string;
  budget: Budget;
}
